security-jwt/src/main/java/io/micronaut/security/token/jwt/validator/JwtTokenValidator.java

```diff
@@ -67,6 +71,7 @@ public JwtTokenValidator(JwtValidator<T> validator,
                              JwtAuthenticationFactory jwtAuthenticationFactory) {
         this.validator = validator;
         this.jwtAuthenticationFactory = jwtAuthenticationFactory;
+        this.scheduler = Schedulers.boundedElastic();
```

Comment:
Would it be better to inject an executor service instead: 

```java
@Named(TaskExecutors.BLOCKING) ExecutorService executorService) { 
...
        this.scheduler = Schedulers.fromExecutorService(executorService);
```

Reply:
I spent probably too much time debating this exact thing with myself. :) As alluded to in the PR description, I feel like the better long-term fix would be to refactor `JwtValidator` and on down the chain through `JwksSignature` to be non-blocking, returning Publisher instead of Optional, etc. With that in mind, my intention was to avoid having to deprecate this constructor in favor of one with the injected Executor since a non-blocking version of the API wouldn't require it. 

I suppose instead of having to deprecate the existing one(s) now, we could add a version with the Executor parameter as an option and mark that one with @Inject. But then we'd probably end up marking that version as deprecated later once a non-blocking API were in place.

Reply:
resolved via https://github.com/micronaut-projects/micronaut-security/pull/1641/commits/2d883750166a49493583365871af568b0aa517ff